pkg/controller/jobs/job/job_webhook.go

```diff
@@ -161,8 +162,7 @@ func validatePartialAdmissionUpdate(oldJob, newJob *Job) field.ErrorList {
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "parallelism"), "cannot change when partial admission is enabled and the job is not suspended"))
 		}
 	}
-
-	if oldJob.IsSuspended() == newJob.IsSuspended() && !newJob.IsSuspended() && oldJob.syncCompletionWithParallelism() != newJob.syncCompletionWithParallelism() {
+	if oldJob.IsSuspended() == newJob.IsSuspended() && !newJob.IsSuspended() && oldJob.syncCompletionWithParallelism() != newJob.syncCompletionWithParallelism() && kubeversion.LessThan(kubeversion.KubeVersion1_27) {
```

Comment:
If so, the `syncCompletionAnnotationsPath` annotation should be made  immutable.

Also this looks like it should be supported only for indexed jobs,
https://github.com/kubernetes/kubernetes/blob/05b64c6b5e11128f7aa720473a60dfd61946cf6a/pkg/apis/batch/validation/validation.go#L570-L597

Additionally maybe we should document the feature gate this is guarded in k/k

https://github.com/kubernetes/kubernetes/blob/41da26dbe15207cbe5b6c36b48a31d2cd3344123/pkg/registry/batch/job/strategy.go#L184.

Reply:
> If so, the `syncCompletionAnnotationsPath` annotation should be made immutable.

It might not be a very important scenario, but after an upgrade to 1.27, a user may want to add the annotation. In that case verifying in `validateCreate` should be enough, letting the annotation be mutable as long as it conforms to the conditions.

> Also this looks like it should be supported only for indexed jobs,

Indeed, for non-indexed jobs `completions` is immutable, so IIUC we should not let setting the annotation `=true` for indexed jobs.

Reply:
For simplicity, I agree that it's better to just make the annotation immutable.

Reply:
sgtm, although IIUC this will require an extra check, but it might be conceptually simpler indeed.